resources/public/pamela.ebnf

```diff
@@ -114,25 +119,36 @@ number-ref = ( symbol | lvar-ctor | number )
 lvar-ctor = ( <LP> <LVAR> <RP> | <LP> <LVAR> string lvar-init? <RP> )
 lvar-init = literal | bounds-literal
 
-pclass-ctor = <LP> pclass-name pclass-ctor-arg* pclass-ctor-option* <RP>
-pclass-name = !MODE_OF symbol
+pclass-ctor = <LP> symbol pclass-ctor-arg* pclass-ctor-option* <RP>
 pclass-ctor-arg = ( boolean | string | number | symbol | pclass-arg-keyword )
 pclass-ctor-option = ( id | plant-part | interface )
 pclass-arg-keyword = !reserved-pclass-ctor-keyword keyword
-reserved-pclass-ctor-keyword = ( <ID> | <INTERFACE> | <PLANT_PART> )
+reserved-pclass-ctor-keyword = ( <ID> | <INTERFACE> | <PLANT_PART> ) stop-token
 id = <ID> string
 plant-part = <PLANT_PART> string
 interface = <INTERFACE> string
 
 bounds = bounds-literal | lvar-ctor
 bounds-literal = <LB> number ( number | INFINITY ) <RB>
 
-symbol = !boolean #'[A-Za-z][A-Za-z0-9-_]*'
+symbol = !reserved-symbol #'[A-Za-z][A-Za-z0-9-_]*'
+reserved-symbol = reserved-string stop-token
+reserved-string = ( <AND> | <ASK> | <ASSERT> | <BETWEEN> | <BETWEEN-ENDS> |
```

Comment:
In the pamela.ebnf, most of the reserved word tokens are in snake_case (e.g., <CHOOSE_WHENEVER>).  However a few of them are in kabab-case (e.g., <BETWEEN-ENDS>).  Since we're doing some housecleaning here, let's make these consistent.